eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -21,7 +21,7 @@ function Find-Suppressions-Yaml {
   $currentDirectory = Get-Item (Split-Path -Path $fileInSpecFolder)
 
   while ($currentDirectory) {
-    # Support both `suppressions.yaml` and `suppressions.yml`, preferring the former
+    # Support both .yaml and .yml, preferring the former
```

Comment:
Why have this complication? I'd say just make it one or the other and not support both.

Reply:
> The best way to approach this is to be rigorous when creating new files (i.e. use .yaml) and be permissive when accepting input (i.e. allow .yml when you encounter it), possible automatically upgrading/correcting these errors when possible.

https://stackoverflow.com/a/54265316

Reply:
In this repo alone, we have `tspconfig.yaml` (TSP does not support `tspconfig.yml`).

But then our pipelines are named `eng/pipelines/*.yml`.

Reply:
This will ultimately be decided when we refactor to shared code.  My preference:

1. Accept both, prefer `.yaml` (e.g. always use `.yaml` in docs, samples)
2. Accept only `.yaml`
3. Accept only `.yml`

Reply:
I don't have a strong opinion but I generally like to remove ambiguity for our stuff. So I'd prefer we support one or the other for these file types rather than letting folks make an independent decision. For example what if someone puts a .yml and .yaml file in the same directory?

Reply:
+1 for avoiding ambiguity. Our SDK partners also work on `sdk-suppressions.yml` so I'll ask them to change it to `sdk-suppressions.yaml`:

- [Pull Request 524197](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/524197): [pipeline-bot] Add the new SDK suppression labels; add support for updating them if sdk-suppressions.yml file was modified

Reply:
Related issue in TypeSpec, which currently only supports `.yaml`:

https://github.com/microsoft/typespec/issues/2257